### PR TITLE
[WIP] Start Gate Mission

### PR DESCRIPTION
**Pull request by jaxnb**
WIP
This is the start gate mission.

Known issues:
* Needs to differentiate between entering and exiting
* Needs real documentation

### CODE REVIEW

File: mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    @util.cancellableInlineCallbacks
+    def init(cls):
+        cls.tfBuffer = tf2_ros.Buffer()
+        cls.listener = tf2_ros.TransformListener(cls.tfBuffer)
+        cls.enablePingerTracking = False
+        cls.intersect_vectors = np.array([])
+
+        cls.pinger_pos_sub = cls.nh.subscribe("/hydrophones/position", PointStamped)
+        cls.heading_sub = rospy.Subscriber('/hydrophones/direction', Vector3Stamped, cls.heading_cb, queue_size=10)
+        cls.alarm_broadcaster = yield TxAlarmBroadcaster.init(cls.nh, 'kill')
```
Comment (ironmig) [inline mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py]: no longer needed

File: mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        scan_lookat_1 = scan_points_1[1]
+        scan_point_2 = scan_points_2[0]
+        scan_lookat_2 = scan_points_2[1]
+
+        # This is necessary because we manually adjust the alarms. Don't want to revive a killed robot.
+        assert not self.killed, 'Cannot run when killed'
+        yield self.hydrophones.disable()
+        yield self.hydrophones.reset()
+
+        print(' 4')
```
Comment (ironmig) [inline mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py]: Best to remove these debug lines. Useful feedback can be sent with `send_feedback`

File: mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+        #print('4')
+
+        #pinger_pos = yield self.get_pinger_pos()
+        #print(pinger_pos)
+
+        #print('5')
+
+        #yield self.hydrophones.disable()
+        # yield self.enable_autonomous()
```
Comment (ironmig) [inline mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py]: Remove dead code

File: mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                bucket1 = 0
+                bucket2 = 0
+                for pinger_vector in intersect_vectors:
+                    t, s = np.linalg.solve(np.array([gate_vector[1]-gate_vector[0], pinger_vector[0]-pinger_vector[1]]).T, pinger_vector[0]-gate_vector[0])
+                    intersection = (1-t)*gate_vector[0] + t*gate_vector[1]
+                    gateDist0 = np.norm(intersection - np.array(gate_results[0][0]))
+                    gateDist1 = np.norm(intersection - np.array(gate_results[0][1]))
+                    gateDist2 = np.norm(intersection - np.array(gate_results[0][2]))
+                    if gateDist0 < gateDist1 and gateDist0 < gateDist2:
+                        bucket0 = bucket0 + 1
```
Comment (ironmig) [inline mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py]: `+= 1`

File: mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+        defer.returnValue(traversal_points_np)
+
+    @util.cancellableInlineCallbacks
+    def heading_cb(self, p_message):
+        if self.enablePingerTracking:
+            try:
+                transformed_vec = self.tfBuffer.transform(p_message,
+                        self.global_frame, rospy.Duration(2))
+                transformed_origin = self.tfBuffer.lookup_transform(self.global_frame,
```
Comment (ironmig) [inline mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py]: Hmmm would be surprised if this didn't require a yield... be sure to test mission in sim to check little things like this

**Comment by jaxnb** [comment]
Messed up some git stuff see #448

**Pull request closed without merging**